lib/amqp/gen/connection_manager.ex

```diff
@@ -65,13 +64,15 @@ defmodule Amqpx.Gen.ConnectionManager do
 
   def terminate(_, %__MODULE__{connection: connection}) do
     if Process.alive?(connection.pid) do
-      Process.exit(connection.pid, :kill)
+      Connection.close(connection)
```

Comment:
I saw that `Connection.close` could return an `error`. It's true that here we are inside a `terminate` function and there's not so much we can do, but maybe it could be worthwhile to add a log for that scenario, WDYT?